scripts/check-backwards-compatability.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+# fetch minimum version and enumerate tags for supported releases
+minimum="v$(curl -s https://semgrep.dev/api/check-version | jq -r '.versions.minimum')"
+tags=$(git log --simplify-by-decoration --pretty=format:%D "${minimum}^!" origin/main | grep -o 'tag: [^,)]\+' | sed 's/^tag: //' | sort -n)
+
+# check tags
+checked=()
+errors=0
+for tag in $tags; do
+    commit=$(git rev-list -n 1 "$tag")
+    if [[ "${checked[*]}" =~ $commit ]]; then
+        echo "Skipping $tag because commit $commit has already been checked"
+        continue
+    fi
+    checked+=($commit)
```

Comment:
here and above, I would quote `"$commit"` because I don't want to trust how it was obtained and more importantly `$commit` should be read as "please split this string".